### ISSUE TITLE
Zarf init support external registry but doesn't accept any insecure flag for registry

### ISSUE THREAD

**Issue by rishi-anand** [defenseunicorns/zarf#1190]
Context
We are trying to leverage an external registry that has a self-signed certificate while deploying package via zarf but while doing zarf init, zarf-agent images doesn't gets uploaded to registry and it throws 'Certifficate Signed by unknown authority'. Also zarf init doesn't accept --insecure flag for registry.
In version 0.22.0 zarf init operation with internal registry wasn't failing even if image push fails. This seems to be a new behavior in v0.23.0, possibly because refactor in image push changes, which is the right behavior.

![image](https://user-images.githubusercontent.com/9067666/211942285-a5161244-4249-4bd2-93f6-d3c6f061ec14.png)

### Environment
Device and OS: MacOS, Ubuntu22.04
App version: 0.23.3
Kubernetes distro being used:
Other:

### Steps to reproduce

- Zarf init using an external registry `zarf init --registry-push-password=abcd --registry-push-username=abcd --registry-url=abcd.registry.local` and abcd.registry.local is a local registry with the self-signed certificate.
- zarf-init fails in 0.23.0 with error

### Expected result
Images in init package should get uploaded to external registry and zarf init should get completed successfully.

### Actual Result
```
  ⠼  Updating image ghcr.io/defenseunicorns/zarf/dev-agent:e32f41ab50f994302614adf62ab6f13a7ecfbb25 (1s)
     ERROR:  unable to deploy all components in this Zarf Package: unable to deploy component zarf-agent: unable to deploy component zarf-agent: unable to push images to the registry: Get "https://harbor-ehl.spectrocloud.com/v2/": x509: certificate signed by unknown authority
root@rishi-vm:~/work/zarf#
```

### Visual Proof (screenshots, videos, text, etc)
![image](https://user-images.githubusercontent.com/9067666/211942516-d429155c-4493-4c0c-9622-b228676d8e82.png)

### Severity/Priority
4 (on scale of 1-5)

### Additional Context
Building a zarf package using images from a self-signed registry also fails. There are open issues https://github.com/defenseunicorns/zarf/issues/561 and https://github.com/defenseunicorns/zarf/issues/1015 for creating package from self signed repo.

**Comment by rishi-anand** [comment]
**_Update:_** I have added support for insecure flag in zarf init. I will raise PR very shortly for review. 
Now, I am facing a **regression** issue when image is getting pushed to the registry. Registry replacement isn't happening while pushing image to the external registry.

![image](https://user-images.githubusercontent.com/9067666/211943654-1d2fbc67-700c-4f9d-82c2-485856bba4b4.png)

ImgConfig has information of external registry but while pushing it is trying to push to default registry without registry replacement. This issue is still reproducible in v0.23.3 which was released yesterday. **Probably external registry support is completely broken in v0.23.0 to v0.23.3 as image can't be pushed to the external repo**.

**Comment by rishi-anand** [comment]
_**Update:**_ Found the root cause of regression issue, It seems assigning external registry url (https://github.com/defenseunicorns/zarf/blob/v0.22.2/src/internal/images/push.go#L25) is missing in v0.23.3.

Will create PR with the fix.

**Comment by Racer159** [comment]
Fixed in #1280
